sympy/physics/mechanics/_pathway.py

```diff
@@ -0,0 +1,273 @@
+    LinearPathway(origin=pO, insertion=pI)
+
+    The pathway create above isn't very interesting without the positions and
+    velocities of its origin and insertion points being described. Without this
+    its not possible to describe how the pathway moves, i.e. its length or its
+    shortening velocity.
+
+    >>> from sympy.physics.mechanics import ReferenceFrame
+    >>> from sympy.physics.vector import dynamicsymbols
+    >>> N = ReferenceFrame('N')
+    >>> q = dynamicsymbols('q')
+    >>> insertion.set_pos(origin, q * N.x)
+    >>> insertion.pos_from(origin)
+    q(t)*N.x
+
+    A pathway's length can be accessed via its ``length`` attribute.
+
+    >>> linear_pathway.length
+    sqrt(q(t)**2)
+
+    Note how what appears to be an overly-complex expression is returned. This
+    is actually required as it ensures that a pathway's length is always
+    positive.
+
+    A pathway's shortening velocity can be accessed similarly via its
+    ``shortening_velocity`` attribute.
+
+    >>> linear_pathway.shortening_velocity
+    -q(t)*Derivative(q(t), t)/sqrt(q(t)**2)
+
+    Parameters
+    ==========
+
+    origin : Point
+        The ``Point`` from which the pathway originates.
+    insertion : Point
+        The ``Point`` into which the pathway inserts.
+
+    """
+
+    def __init__(
+        self,
+        *,
+        origin: Point,
+        insertion: Point,
+    ) -> None:
+        """Initializer for ``LinearPathway``.
+
+        Parameters
+        ==========
+
+        origin : Point
+            The ``Point`` from which the pathway originates.
+        insertion : Point
+            The ``Point`` into which the pathway inserts.
+
+        """
+        super().__init__(
+            origin=origin,
+            insertion=insertion,
+        )
+
+    def _true_length(self) -> Expr:
+        """Exact analytical expression for the pathway's length."""
+        length = self.insertion.pos_from(self.origin).magnitude()
+        return length
+
+    def _true_shortening_velocity(self) -> Expr:
+        """Exact analytical expression for the pathway's shortening velocity."""
+        relative_position = self.insertion.pos_from(self.origin)
+        if not relative_position:
+            return S.Zero
+        t = dynamicsymbols._t  # type: ignore
+        # A reference frame is needed to differentiate ``relative_position`` to
+        # ``relative_velocity`` so choose the first ``ReferenceFrame`` that
+        # ``relative_position`` is defined using.
+        frame = relative_position.args[0][1]
+        relative_velocity = relative_position.diff(t, frame)
+        shortening_velocity = -relative_velocity.dot(relative_position.normalize())
+        return shortening_velocity
+
+    def _unit_vector_to_origin(self) -> Vector:
+        """Unit vector towards the pathway's origin from its insertion."""
+        return self.origin.pos_from(self.insertion) / self.length
+
+    def _unit_vector_to_insertion(self) -> Vector:
```

Comment:
Not sure whether I would consider this one a duplicate from the previous, as it is just a `-` difference.

Reply:
These methods could be inlined without losing readability.

Reply:
Addressed in 234ba42d3ce3b823956100aa1b9056acc83720d6.